Gems/ROS2/Code/Source/ROS2ModuleInterface.h

```diff
@@ -61,7 +62,7 @@ namespace ROS2
                   AckermannControlComponent::CreateDescriptor(),
                   RigidBodyTwistControlComponent::CreateDescriptor(),
                   SkidSteeringControlComponent::CreateDescriptor(),
-                  ROS2CameraSensorComponent::CreateDescriptor(),
+                  ROS2CameraSensorEditorComponent::CreateDescriptor(),
```

Comment:
This is the wrong file to add the new editor component's descriptor, this file is for runtime components/systems, it needs to be added to [ROS2EditorModule.cpp](https://github.com/o3de/o3de-extras/blob/fec2177d8683913c35bb544fea30d398fc4c00a3/Gems/ROS2/Code/Source/ROS2EditorModule.cpp#L28) which has the editor components/systems.

I noticed something else that seems to be wrong:
- The [editor shared target](https://github.com/o3de/o3de-extras/blob/fec2177d8683913c35bb544fea30d398fc4c00a3/Gems/ROS2/Code/CMakeLists.txt#L116) is including `ros2_editor_files.cmake` and it shouldn't, because it's the editor static target the one that includes it. The editor shared target should only include `ros2_editor_shared_files.cmake`.